### PR TITLE
Many changes to vo_opengl, mostly concerning scalers

### PR DESCRIPTION
**Pull request by haasn**
The amount of unmerged commits piling up in my repository is getting a bit out of hand. Many users have requested to me that this stuff land back in mpv at some point, so I'm willing to cooperate to help make that possible for the benefit of everybody.

I expect/understand that most of this probably needs to be refactored hard, and some shit (like loop unrolling) is probably not worth including at all. Up to you, I just wanted to create an issue for it to at least get something in motion.

tl;dr this is the roadmap of changes I would love to make to mpv in my personal dream world

### CODE REVIEW

**Comment by wm4** [comment]
Most commits ok, but rebase your branch and remove the merge commits before we proceed.

**Comment by haasn** [comment]
Updated/rebased/adjusted, and also got rid of scale-sep/indirect.

Still undecided about d6e1ef5, if you don't want to include it I'll drop that from the PR.

**Comment by wm4** [comment]
Seems like we just decided on IRC to remove them.

But maybe still add a notice to the manpage how to get some of the old scalers?

**Comment by wm4** [comment]
Other than whatever I commented on, looks ok.

**Comment by wm4** [comment]
>(Reinitializing the scale_sep_fbo is not needed of course because bilinear etc. don't ever use it)

What if you switch from bilinear to something that needs it? Also I don't see where the "indirect" FBO is initialized?

**Comment by haasn** [comment]
(As discussed on IRC, but for completeness' sake:) By https://github.com/haasn/mpv/blob/a116cbe46fbf5ce51da0304d0abaa3323eebc2e3/video/out/gl_video.c#L1513

**Comment by haasn** [comment]
Merged. Glad we could finally get this stuff resolved.
